tools/rpp/src/lib.rs

```diff
@@ -172,8 +183,11 @@ pub fn process_str(s: &str, ctx: &mut Context) -> Result<String, ParsingError> {
     });
 
     let mut out = String::new();
-    for line in no_comments.lines() {
-        ctx.line_num += 1;
+
+    // Collect range of splits
+    let split_lines_range: Vec<_> = ctx.split_lines_re.find_iter(&no_comments).collect();
```

Comment:
What if you just did a find/replace of line continuations (lines ending in `\`) and then do the same as what's done for block comments (above). Then you shouldn't need any of the other changes in this file.

Reply:
This was used so as to keep track of the correct line number. If lines ending in `\` are escaped here, wouldn't that mess up the line number in process_line?